src/parse_execution.rs

```diff
@@ -567,6 +568,19 @@ impl<'a> ParseExecutionContext {
                 "The expanded command was empty."
             );
         }
+        // Complain if we've expanded to a subcommand keyword like "command" or "if".
+        // This is an attempt to defeat function resolution.
+        //
+        // Make an exception for "time" because that is frequently used as a command and does fundamentally the same thing.
+        if parser_keywords_is_subcommand(out_cmd) && !unexp_cmd.starts_with(out_cmd.chars()) && out_cmd != L!("time") {
```

Comment:
I'm confused by `!unexp_cmd.starts_with(out_cmd.chars())`. Given

    set cmd command 1 2 3
    $cmd

`out_cmd` will be `command` (the args are put in a separate variable).
So we should be able to use `unexp_cmd != out_cmd`, right?
A test case would be `command""` - it expands to `command` and we probably want to report this error.
Unless this is only about variable expansion?

Reply:
This was an attempt at a very cheesy check for "there was some expansion".

I.e. we do allow a literal `command foo`, we don't allow `set -l var command; $var foo`.

I'll have to check if unexp_cmd is actually *just* the command, I think I might have been confused there.

>A test case would be command"" - it expands to command and we probably want to report this error.
Unless this is only about variable expansion?

`command"" echo foo` currently uses `command` as a decorator and works. I don't see a problem with keeping that, the problem is really *just* with things that expand and are understood to be the command *builtin*.

Reply:
> I'll have to check if unexp_cmd is actually just the command

yeah it is, it's `let unexp_cmd = self.node_source(&statement.command);`  and `statement.command` is a singular string token in command position